code/modules/reagents/reagent_containers/food/snacks/noodles.dm

```diff
@@ -170,3 +170,29 @@
 	reagents_to_add = list(/singleton/reagent/nutriment = 8)
 	reagent_data = list(/singleton/reagent/nutriment = list("pasta" = 5, "pumpkin squash" = 5, "creamy sauce" = 5))
 	filling_color = "#80d1e9"
+
+/obj/item/reagent_containers/food/snacks/ravioli/nakarka
+	name = "nakarka ravioli"
+	icon_state = "raviolinakarka"
+	desc = "Ravioli with Nakarka cheese filling in a sour cream and chives based sauce."
+	reagents_to_add = list(/singleton/reagent/nutriment = 10)
+	reagent_data = list(/singleton/reagent/nutriment = list("sharp tanginess" = 5, "pasta" = 4, "sour cream" = 4, "chives" = 3))
+
+/obj/item/reagent_containers/food/snacks/fettuccine_alfredo
+	name = "fettuccine alfredo"
+	desc = "Pasta lathered in a white, buttery parmesan cheese and cream sauce. Classic Italian!"
+	icon = 'icons/obj/item/reagent_containers/food/noodles.dmi'
+	icon_state = "alfredo"
+	trash = /obj/item/trash/plate
+	filling_color = "#f8e8b1"
+	center_of_mass = list("x"=16, "y"=10)
+	reagents_to_add = list(/singleton/reagent/nutriment = 6, /singleton/reagent/nutriment/protein/cheese = 2)
+	reagent_data = list(/singleton/reagent/nutriment = list("pasta" = 5, "garlic" = 3), /singleton/reagent/nutriment/protein/cheese = list("creamy parmesan" = 5))
+
+/obj/item/reagent_containers/food/snacks/fettuccine_alfredo/update_icon()
+	var/percent_fettuccine_alfredo = round((reagents.total_volume / 8) * 100)
```

Comment:
The percentage have to be calculated based on the total amount of reagents this takes, not against a fixed number that might not be updated if the reagents change

Reply:
In all of these the number chosen is based on the total amount of reagents in the dish. It's the way it's coded in every other existing dish in the game that changes appearance based on how much of it is eaten dating back to the steak. I tested it works, the only thing that might throw it out of whack is if someone adds a significant amount of reagents to the dish such as pouring large amounts of liquid onto it and if there is a way to account for that in code i don't really know what that would be so I formatted it based on existing dishes while changing the number total to the number of reagents in a dish as it is when the dish is dcreated.

Reply:
I don't see it done anywhere else like this by a quick look

You can calculate the total amount of reagents in the dish by looping the reagents_to_add list and adding up the values, or by `.maximum_volume`, depending on what you want to do exactly

Reply:
calculating the total amount of reagents in the dish by adding the reagents_to_add list is exactly what i did and how i got the numbers i put in. Unless i am mistaken they should all have the number of reagents in them as the divisable number.

also if you are looking for other dishes that use this code there is steak, xenosteak, neaera brain loaf, packed rice bowl, as well as other dishes i created previously such as lasgna slices, floating island, north 60 squid, meatballs and peas, schnitzel, and others.

i'm not sure i understand what the change you are suggesting is. would "var/percent = round((reagents.total_volume / volume) * 100)" work better? (I don't know how to write brand new code but that's how the code for toothpaste looks)

Reply:
>calculating the total amount of reagents in the dish by adding the reagents_to_add list is exactly what i did and how i got the numbers i put in

You did a manual calculation, I'm saying it should do it dinamically by traversing the list, so when/if the reagents are changed there would be no need to update the formula or risk of inconsistency if someone forgets to

Reply:
i see.
still no clue how to do that though. will replacing it with "var/percent = round((reagents.total_volume / volume) * 100)" fix it?

Reply:
marking as resolved as this dish has been fixed according to fluffyghoster's request and suggestions